torch/onnx/_internal/exporter.py

```diff
@@ -960,6 +953,12 @@ def _assert_fake_tensor_mode(self):
                 and not isinstance(x, torch._subclasses.FakeTensor),
                 (self.model.parameters(), self.model.buffers()),
             )
+        if (
+            has_any_fake_tensor or has_any_fake_param_or_buffer
+        ) and not self.options.fake_context:
+            raise RuntimeError(
```

Comment:
Related to the comment above, I suppose this should still raise for that case?

Reply:
Yes, I could move it back. I just found it's hard to pdb it when it's separate so far lol.